### ISSUE TITLE
More descriptive error message when two snippets have the same name

### ISSUE THREAD

**Issue by wuffi** [rstudio/rmarkdown#1276]
Whenever two snippets have the same name (identifier), it is not possible to "knit" any output, which technically is the way it should be.

The problem is the resulting error message, which is difficult to interpret/understand. 
A more descriptive one would help a lot in identifying duplicate snippet names, which can easily arise from copying and pasting code.

**Comment by mikey-harper** [comment]
I agree that this could be a useful feature.  Error messages are given for other common formatting errors. For example, if you forget quote your captions: 

````
```{r cars, echo = TRUE, fig.cap = This is my caption}
plot(cars)
```
````
> NOTE: I saw chunk options "cars, echo = TRUE, fig.cap = ""
 please go to https://yihui.name/knitr/options
 (it is likely that you forgot to quote "character" options)

 @yihui , the function `parse_block` could be updated:

```
else stop("duplicate label '", label, "'.  Please make sure all code chunks have unique labels. \n(It is likely that you have copied and pasted a chunk)".)
```
There is the option to set `options(knitr.duplicate.label = 'allow')` but as you said when releasing this, it should only be used if you know the full consequences.  So I wouldn't make any mention of this setting in the printout.

**Comment by wuffi** [comment]
Thanks a lot for the quick reply. It is certainly not an urgent issue, but would improve usability.
The `knitr.duplicate.label` option is certainly very nice to know!
